connector/connector_ldap.go

```diff
@@ -217,11 +250,108 @@ func (c *LDAPConnector) TrustedEmailProvider() bool {
 	return c.trustedEmailProvider
 }
 
+// A LDAPPool is a Connection Pool for LDAP connections
+// Initialize exported fields and use Get() to get a connection.
+// Use Put() to put it back into the pool.
+type LDAPPool struct {
+	m              sync.Mutex
+	conns          map[*ldap.Conn]struct{}
+	MaxIdleConn    int
+	PoolCheckTimer time.Duration
+	ServerHost     string
+	ServerPort     uint16
+	UseTLS         bool
+	UseSSL         bool
+	TlsConfig      *tls.Config
+}
+
+func (p *LDAPPool) Get(params ...*ldap.Conn) (*ldap.Conn, error) {
```

Comment:
Can you separate getting a random connection (Get()) and getting a specific connection (Get(params)) into different methods? They have very different uses and it's odd to overload this method.